PurchasesCoreSwift/Purchasing/PurchasesOrchestrator.swift

```diff
@@ -0,0 +1,506 @@
+        productsManager.products(withIdentifiers: productIdentifiersSet) { products in
+            self.operationDispatcher.dispatchOnMainThread {
+                completion(Array(products))
+            }
+        }
+    }
+
+    @available(iOS 12.2, macOS 10.14.4, watchOS 6.2, macCatalyst 13.0, tvOS 12.2, *)
+    @objc public func paymentDiscount(forProductDiscount productDiscount: SKProductDiscount,
+                                      product: SKProduct,
+                                      completion: @escaping (SKPaymentDiscount?, Error?) -> Void) {
+        receiptFetcher.receiptData(refreshPolicy: .onlyIfEmpty) { maybeReceiptData in
+            guard let receiptData = maybeReceiptData,
+                  !receiptData.isEmpty else {
+                      completion(nil, ErrorUtils.missingReceiptFileError())
+                      return
+                  }
+            guard let discountIdentifier = productDiscount.identifier else {
+                      completion(nil, ErrorUtils.productDiscountMissingIdentifierError())
+                      return
+                  }
+            self.backend.post(offerIdForSigning: discountIdentifier,
+                              productIdentifier: product.productIdentifier,
+                              subscriptionGroup: product.subscriptionGroupIdentifier ,
+                              receiptData: receiptData,
+                              appUserID: self.appUserID) { maybeSignature, maybeKeyIdentifier, maybeNonce, maybeTimestamp, maybeError in
+                if let error = maybeError {
+                    // TODO: Consider making a custom error.
+                    completion(nil, error)
+                    return
+                }
+
+                guard let keyIdentifier = maybeKeyIdentifier,
+                      let nonce = maybeNonce,
+                      let signature = maybeSignature,
+                      let timestamp = maybeTimestamp else {
+                          // TODO: Consider making a custom error.
+                          completion(nil, ErrorUtils.unexpectedBackendResponseError())
+                          return
+                      }
+
+                let paymentDiscount = SKPaymentDiscount(identifier: discountIdentifier,
+                                                        keyIdentifier: keyIdentifier,
+                                                        nonce: nonce,
+                                                        signature: signature,
+                                                        timestamp: timestamp)
+                completion(paymentDiscount, nil)
+            }
+        }
+    }
+
+    @objc public func purchase(product: SKProduct,
+                               payment: SKMutablePayment,
+                               presentedOfferingIdentifier maybePresentedOfferingIdentifier: String?,
+                               completion: @escaping PurchaseCompletedBlock) {
+        Logger.debug("makePurchase")
```

Comment:
this log doesn't feel very helpful, idk if we want to add more info? or at least say `makePurchase called` for clarity

Reply:
Totally agree, this was the old message, but it's not super helpful, updating.